cards.js

```diff
@@ -17,7 +17,116 @@ function shuffle(array){
 }
 
 function buildDeck(){
+  var deck = [];
+  for (var i = 0; i < suits.length; i++) {
+    for (var j = 0; j < values.length; j++) {
+      deck.push({ suit: suits[i], value: values[j]});
+    };
+  };
+  return deck;
+}
+
+var newDeck = buildDeck();
+shuffle(newDeck);
+
+function faceCardValue(value){
+  if(value === 'J'){
+    return 11;
+  } else if(value === 'Q'){
+    return 12;
+  } else if(value === 'K'){
+    return 13;
+  } else if(value === 'A'){
+    return 14;
+  } else{
+    return value;
+  }
+}
+
+function printCard(card){
+  return card.value + ' ' + card.suit;
+}
+
+function highCard(deck){
+  var card1 = deck.shift();
+  var card2 = deck.shift();
+  var winner;
+  if (faceCardValue(card1.value) > faceCardValue(card2.value)){
+    winner = card1;
+  } else if(faceCardValue(card1.value) < faceCardValue(card2.value)){
+    winner = card2;
+  };
+  console.log('card 1: ' + printCard(card1));
+  console.log('card 2: ' + printCard(card2));
+  if (winner === undefined) {
+    console.log('It\'s a tie');
+  } else{
+    console.log(printCard(winner) + ' won');
+  };
+}
 
-// Put some code in here!
+while(newDeck.length > 0){
+  highCard(newDeck);
+};
 
+var secondDeck = buildDeck();
+shuffle(secondDeck);
+
+function gameOfWar(deck){
+  var turnsTaken = 0;
+  var p1Deck = [];
+  var p2Deck = [];
+  while(deck.length > 0){
+    p1Deck.push(deck.shift());
+    p2Deck.push(deck.shift());
+  };
+  while(p1Deck.length > 0 && p2Deck.length > 0){
```

Comment:
As written, as soon as `p1Deck.length === 0`, the loop will stop regardless of `p2Deck.length`. If that's a problem, you could use a logical OR instead: `while(p1Deck.length > 0 || p2Deck.length > 0){`. Alternatively, if the decks are depleted at the same rate, you could just check for either `p1Deck.length` or `p2Deck.length`, exclusively.